sycl/include/sycl/ext/intel/experimental/esimd/detail/esimd_types.hpp

```diff
@@ -10,11 +10,12 @@
 
 #pragma once
 
-#include <CL/sycl/INTEL/esimd/detail/esimd_region.hpp>
-#include <CL/sycl/INTEL/esimd/esimd_enum.hpp>
 #include <CL/sycl/detail/defines.hpp>
 #include <CL/sycl/detail/stl_type_traits.hpp> // to define C++14,17 extensions
 #include <CL/sycl/half_type.hpp>
+#include <sycl/ext/intel/experimental/esimd/detail/esimd_region.hpp>
+#include <sycl/ext/intel/experimental/esimd/esimd_enum.hpp>
+
 #include <cstdint>
 
 __SYCL_INLINE_NAMESPACE(cl) {
```

Comment:
>> Not using CL as the top directory, as top-level cl namespace
has been deprecated in SYCL2020, so top-level CL include directory
will eventually be removed.

No strong objections, but could you please clarify what is the motivation to move only ESIMD related headers? Probably it would be better to move all the headers in one patch.

>> Header location must reflect the contained API's namespace.

It seems this movement doesn't fully achieve that: namespace - path has no `cl` while namespace has.

Reply:
> Probably it would be better to move all the headers in one patch.

As discussed with @pvchupin, there are different complications with moving other headers which need to be solved. E.g. INTEL->intel and ONEAPI->oneapi namespace conversion etc. So it seems like a separate big work.

> It seems this movement doesn't fully achieve that: namespace - path has no cl while namespace has.

I tried to explain the motive in the comments, may I be wasn't clear enough. My point is that eventually CL will be removed (2020 spec does not have it), so moving headers straight to the final place seems a better choice. - ?